src/main/java/apoc/refactor/GraphRefactoring.java

```diff
@@ -299,16 +299,30 @@ public void categorize(
     }
 
     private Node mergeNodes(Node source, Node target, boolean delete) {
-        copyRelationships(source, copyProperties(source, copyLabels(source, target)), delete);
-        if (delete) source.delete();
+        Map<String, Object> propTemp = new HashMap<>();
+        List<String> keys = new ArrayList<>();
+        for(Map.Entry<String, Object> prop : source.getAllProperties().entrySet()){
+            propTemp.put(prop.getKey(), prop.getValue());
+            keys.add(prop.getKey());
+            source.removeProperty(prop.getKey());
+        }
+        copyRelationships(source, copyLabels(source, target), delete);
+        setProperty(propTemp, keys, target);
         return target;
     }
 
+    private void setProperty(Map<String, Object> propTemp, List<String> keys, Node target) {
+        for(int i=0;i<propTemp.size();i++) {
+            target.setProperty(keys.get(i), propTemp.get(keys.get(i)));
+        }
+    }
+
     private Node copyRelationships(Node source, Node target, boolean delete) {
         for (Relationship rel : source.getRelationships()) {
             copyRelationship(rel, source, target);
             if (delete) rel.delete();
         }
+        if (delete) source.delete();
```

Comment:
will this break any other code that uses this? I think the original contract for this method was that rels are deleted but the nodes are left alone.
so I'd rather move this out of the method back into mergeNodes where it originally was.